code/code/misc/combat.cc

```diff
@@ -1992,9 +1992,10 @@ int TBeing::getWeaponDam(const TBeing *v, const TThing *wielded, primaryTypeT is
     // 2h Specialization: should be no additional dam for anyone without specialization
     // extra 10% for 2h is part of the object
     TObj *obj = dynamic_cast<TObj *>(heldInPrimHand());
-    if (obj && obj->isPaired() && doesKnowSkill(SKILL_2H_SPEC_DEIKHAN)) {
+    if (obj && obj->isPaired() && (doesKnowSkill(SKILL_2H_SPEC_DEIKHAN) || doesKnowSkill(SKILL_2H_SPEC_WARRIOR))) {
```

Comment:
Check out how SKILL_DUAL_WIELD is handled just above here. Both warriors and thieves have dual wield
- There are two skills SKILL_DUAL_WIELD and SKILL_DUAL_WIELD_THIEF
- getSkillNum() will return the appropriate skill based on learnedness: https://github.com/sneezymud/sneezymud/blob/master/code/code/misc/discipline.cc#L632